ophyd/areadetector/plugins.py

```diff
@@ -632,6 +633,56 @@ class ROIPlugin(PluginBase):
                   doc='Minimum size of the ROI in XYZ',
                   default_read_attrs=('min_x', 'min_y', 'min_z'))
 
+    def set(self, region):
+        ''' This functions allows for the ROI regions to be set.
+
+        This function takes in an ROI_number, and a dictionary of tuples and sets the ROI region.
+
+        PARAMETERS
+        ----------
+        region: dictionary.
+            A dictionary defining the region to be set, which has the structure:
+            { 'x':[min, size], 'y':[min, size], 'z':[min, size]}. 
+            Any of the keywords can be ommitted, and they will be ignored.
+        '''
+        if region is not None:
+            for direction, value in region.items():
+                getattr(self, f'min_xyz.min_{direction}').put(value[0])
```

Comment:
we should use `set` and `&` all of the status object together instead of returning `NullStatus` here.

Reply:
I am not sure what you mean by 'use set and & all of the status objects together', can you elaborate?

Reply:
`NullStatus()` is a cheat -- it says, "I'm done." Using it here would give the RunEngine false information: the settings above have not necessarily been processed.

To give the RunEngine accurate information, we normally replace:

```
signal.put(val)
return NullStatus()
```

with

```
status = signal.set(val)
return status
```

but here because you are setting multiple things, and we want to know when they are _all_ done, we can use the fact that status objects can be logically AND-ed like `status1 & status2 & status3`. To do this for an arbitrary number of status objects you'll need `functools.reduce` and `operator.and_`.